### PR TITLE
Add JDK 11 to CI test matrix

### LINKED ISSUE #320: Default CI Java Version to Java 11, run tests on 8, 14 and 17
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/64, default CI Java version to 11 (LTS), and run tests on 8, 14 and 17.

- [ ] Change the default Java version for CI in GitHub actions from 14/15 to 11 (LTS).
- [ ] Add Java 8, 14 (used before), and 17 (next LTS) to the CI matrix.
- [ ] Document Java versions in DEVELOPER_GUIDE.md.

### PR DESCRIPTION
**Pull request by ohltyler**
Signed-off-by: Tyler Ohlsen <ohltyler@amazon.com>

### Description
One of the campaigns for upcoming 1.3 release is testing on JDK 11, with optional versions of 8, 14, 17. This updates the workflow to add JDK 11. There are other problems with 8/17 so ignoring those for now.

Also removes CD workflow since that hasn't been used since pre-OpenSearch days.
 
### Issues Resolved
Resolves #320 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by amitgalitz — APPROVED** [review]
LGTM, just make sure to wait until CI passes especially as this involves CI workflow changes. Also JDK8 might not be working because of protostuff dependency compiled with java 11, before that change was pushed it in, it was okay to use runtime jdk8. However the exception itself mentions something was compiled with java version 54 which is equivalent to java 10 so not 100% sure. Can be addressed in future PRs

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -8,11 +8,10 @@ on:
       - "*"
 
 jobs:
-
   Build-ad:
     strategy:
       matrix:
-        java: [14]
+        java: [11, 14]
```
Comment (ylwu-amzn) [inline .github/workflows/CI.yml:14]: All plugins and OpenSearch core should make sure Java 11 works for 1.3 release?

**Review by ylwu-amzn — APPROVED** [review]

File: .github/workflows/CI.yml (hunk 1 above)
Comment (ohltyler) [reply .github/workflows/CI.yml:14]: Yes

Comment (ohltyler) [reply .github/workflows/CI.yml:14]: 11 is the default used by bundle build / integ tests

**Pull request merged**